ArcadiaParties-SPA/src/app/department/effects/effect.ts

```diff
@@ -1,18 +1,22 @@
 import { createEffect, ofType, Actions } from '@ngrx/effects';
 import { Injectable } from '@angular/core';
-import { map } from 'rxjs/operators';
+import { map, mergeMap } from 'rxjs/operators';
 
 import { DepartmentService } from '../services/department.service';
 import { loadDepartmentsSuccessAction, loadDepartmentsAction } from '../actions/actions';
 
 @Injectable()
 export class DepartmentEffect {
-    loadDepartments = createEffect(() =>
-        this.actions.pipe(
-            ofType(loadDepartmentsAction),
-            map(() => loadDepartmentsSuccessAction({ departments: this.departmentService.getDepartments() }))
-        )
-    );
+  loadDepartments = createEffect(() =>
+    this.actions.pipe(
+      ofType(loadDepartmentsAction),
+      mergeMap(() =>
+        this.departmentService
+          .getDepartments()
+          .pipe(map(departmentsFromService => loadDepartmentsSuccessAction({ departments: departmentsFromService })))
```

Comment:
You don't need to use another pipe, you can do all of this in top-level pipe to reduce nesting.

Reply:
Done according to your advice